src/cluster.c

```diff
@@ -2012,6 +2012,21 @@ int writeHostnamePingExt(clusterMsgPingExt **cursor) {
     return extension_size;
 }
 
+/* Write the forgotten node ping extension at the start of the cursor, update
+ * the cursor to point to the end of the written extension and return the number
+ * of bytes written. */
+int writeForgottenNodePingExt(clusterMsgPingExt **cursor, sds name, uint64_t ttl) {
+    serverAssert(sdslen(name) == CLUSTER_NAMELEN);
+    clusterMsgPingExtForgottenNode *ext = &(*cursor)->ext[0].forgotten_node;
+    memcpy(ext->name, name, CLUSTER_NAMELEN);
+    ext->ttl = htonu64(ttl);
+    uint32_t extension_size = sizeof(clusterMsgPingExt) + sizeof(clusterMsgPingExtForgottenNode);
+    (*cursor)->type = htons(CLUSTERMSG_EXT_TYPE_FORGOTTEN_NODE);
+    (*cursor)->length = htonl(extension_size);
+    *cursor = (clusterMsgPingExt *) (ext->name + sizeof(clusterMsgPingExtForgottenNode));
```

Comment:
Seems like there is a `UndefinedBehaviorSanitizer`, i ran a daily and it report this
```
cluster.c:2053:48: runtime error: index 48 out of bounds for type 'char [40]'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior cluster.c:2053:48 in 
```

Reply:
Thanks. I'm on vacation. Do you mind looking into this? Or anyone else...?

Reply:
i guess it is a false positive, i added a `REDIS_NO_SANITIZE("bounds")` like others, #11053